experimental/dds/tree2/src/shared-tree-core/branch.ts

```diff
@@ -42,15 +42,19 @@ import { TransactionStack } from "./transactionStack";
  * transaction completes and all pending commits are replaced with a single squash commit.
  */
 export type SharedTreeBranchChange<TChange> =
-	| { type: "append"; change: TaggedChange<TChange>; newCommits: readonly GraphCommit<TChange>[] }
+	| {
+			type: "append";
+			change: () => TaggedChange<TChange>;
```

Comment:
Same thing here - if it were a getter, then all the places that _aren't_ doing expensive work below don't need to make lambdas. Only the one that is expensive would implement the property as a getter.